docs/usage/json_schema.md

```diff
@@ -826,4 +715,15 @@ print(json.dumps(Person.model_json_schema(), indent=2))
 """
 ```
 
-Note that you *must* return a schema, even if you are just mutating it in-place.
+Note that you *must* return a schema, even if you are just mutating it in place.
+
+## Customizing the JSON schema generation process
+
+If you need custom schema generation, you can use a `schema_generator`, modifying the
+[`GenerateJsonSchema`][pydantic.json_schema.GenerateJsonSchema] class as necessary for your application.
+
+The various methods that can be used to produce JSON schema accept a keyword argument `schema_generator: type[GenerateJsonSchema] = GenerateJsonSchema`, and you can pass your custom subclass to these methods in order to use your own approach to generating JSON schema.
+
+`GenerateJsonSchema` implements the translation of a type's `pydantic-core` schema into a JSON schema.
+By design, this class breaks the JSON schema generation process into smaller methods that can be easily overridden in
+subclasses to modify the "global" approach to generating JSON schema.
```

Comment:
I think there's an example that we can use from our code source with discriminators that we can give as an example here.